test/jdk/java/lang/instrument/NativeMethodPrefixApp.java

```diff
@@ -22,25 +22,107 @@
  */
 
 
-import java.io.PrintStream;
-import java.util.*;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.lang.management.*;
+
 import bootreporter.*;
+import jdk.test.lib.helpers.ClassFileInstaller;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
 
+/*
+ * @test
+ * @bug 6263319
+ * @summary test setNativeMethodPrefix
+ * @requires ((vm.opt.StartFlightRecording == null) | (vm.opt.StartFlightRecording == false)) & ((vm.opt.FlightRecorder == null) | (vm.opt.FlightRecorder == false))
+ * @modules java.management
+ *          java.instrument
+ * @library /test/lib
+ * @build bootreporter.StringIdCallback bootreporter.StringIdCallbackReporter
+ *        asmlib.Instrumentor NativeMethodPrefixAgent
+ * @enablePreview
+ * @comment The test uses asmlib/Instrumentor.java which relies on ClassFile API PreviewFeature.
+ * @run driver NativeMethodPrefixApp roleDriver
+ */
 public class NativeMethodPrefixApp implements StringIdCallback {
 
     // This test is fragile like a golden file test.
     // It assumes that a specific non-native library method will call a specific
     // native method.  The below may need to be updated based on library changes.
     static String goldenNativeMethodName = "getStartupTime";
 
-    static boolean gotIt[] = {false, false, false};
+    static boolean[] gotIt = {false, false, false};
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 1) {
+            if (!"roleDriver".equals(args[0])) {
+                throw new Exception("unexpected program argument: " + args[0]);
+            }
+            // launch the NativeMethodPrefixApp java process after creating the necessary
+            // infrastructure
+            System.out.println("creating agent jar");
+            final Path agentJar = createAgentJar();
+            System.out.println("launching app, with javaagent jar: " + agentJar);
+            launchApp(agentJar);
+        } else {
+            System.err.println("running app");
+            new NativeMethodPrefixApp().run();
+        }
+    }
+
+    private static Path createAgentJar() throws Exception {
+        // first setup a directory containing (only) the classes needed in boot classpath
+        // of the application jar to be launched
+        final Path bootClassesDir = createBootClassesDir();
+        final Path agentJar = Path.of("NativeMethodPrefixAgent.jar");
+        final String manifest = """
+                Manifest-Version: 1.0
+                Premain-Class: NativeMethodPrefixAgent
+                Can-Retransform-Classes: true
+                Can-Set-Native-Method-Prefix: true
+                """
+                + "Boot-Class-Path: " + bootClassesDir.toString().replace(File.separatorChar, '/') + "/"
```

Comment:
AFAIU Boot-Class-Path is needed only because the agent needs bootreporter.StringIdCallbackReporter
I don't see much point in separate bootreporter dir.
So we can set Boot-Class-Path to System.getProperty("test.classes") and drop createBootClassesDir() logic

Reply:
Hello Alex, I have updated the PR to add "test.classes" to Boot-Class-Path instead of just those 2 classes. Tests in the updated PR continue to pass.